pkg/router/routeapihelpers/validation.go

```diff
@@ -200,6 +216,28 @@ func ExtendedValidateRoute(route *routev1.Route) field.ErrorList {
 		}
 	}
 
+	if len(tlsConfig.Certificate) > 0 {
+		if certBytes, keyBytes, err := splitCertKey([]byte(tlsConfig.Certificate)); err != nil {
+			result = append(result, field.Invalid(tlsFieldPath.Child("certificate"), "redacted key data", err.Error()))
+		} else {
+			if len(tlsConfig.Key) > 0 {
+				if len(keyBytes) > 0 {
+					keyBytes = append(keyBytes, byte('\n'))
```

Comment:
I noticed there isn't a unit test that covers this branch. I point this out because I am slightly confused as to what's going on here. If `tlsConfig.Key` is set and `splitCertKey` yields private `keyBytes` from `tlsConfig.Certificate`, then `keyBytes` will become a new-line delimited buffer of both of the keys? Could you elaborate a little bit so I can understand?

Reply:
Right, we'll move any private parts that are in `tlsConfig.Certificate` will get moved to `tlsConfig.Key`, preserving any existing private parts that were already in `tlsConfig.Key`.  I'm not sure whether it would be better just to ignore `keyBytes` if `tlsConfig.Key` is already set, or override `tlsConfig.Key` with `keyBytes` if the latter is nonempty, but I think this is is the safest approach, keeping in mind that we don't want to break existing routes that may be doing weird things in `tlsConfig`.

Reply:
Understood. I agree this approach sounds like the way to go.